inc/compatibility/wpml.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Holds the compatibility logic for WPML plugin.
+ *
+ * Author:          Bogdan Preda <friends@themeisle.com>
+ * Created on:      17/03/2023
+ *
+ * @package Neve\Compatibility
+ */
+
+namespace Neve\Compatibility;
+
+/**
+ * Class WPML
+ *
+ * @package Neve\Compatibility
+ */
+class WPML {
+
+	/**
+	 * Init function.
+	 *
+	 * @return bool
+	 */
+	public function init() {
+		if ( ! $this->should_load() ) {
+			return false;
+		}
+		$this->load_hooks();
+
+		return true;
+	}
+
+	/**
+	 * Load hooks.
+	 */
+	public function load_hooks() {
+		add_filter( 'neve_filter_featured_posts', array( $this, 'filter_posts_and_return_original' ), 10, 1 );
+	}
+
+	/**
+	 * Decide if this class should load.
+	 *
+	 * @return bool
+	 */
+	private function should_load() {
+		return defined( 'WPML_PLUGIN_PATH' );
+	}
+
+	/**
+	 * This will return the original language posts.
+	 * This hooks into `neve_filter_featured_posts` filter.
+	 * We ensure that returned posts are in the original language.
+	 *
+	 * @param array $posts The posts to filter.
+	 *
+	 * @return array
+	 */
+	final public function filter_posts_and_return_original( $posts ) {
+		$original_post_ids = array();
+		$current_language  = apply_filters( 'wpml_current_language', null );
+		foreach ( $posts as $post ) {
+			$original_post_id = apply_filters( 'wpml_object_id', $post['ID'], $post['post_type'], true, $current_language );
+			if ( $original_post_id !== $post['ID'] ) {
+				$original_post_ids = $this->add_to_array_unique( $original_post_ids, $original_post_id );
+				continue;
+			}
+			$original_post_ids = $this->add_to_array_unique( $original_post_ids, $post['ID'] );
+		}
+
+		return $original_post_ids;
+	}
+
+	/**
+	 * Add to array unique values.
+	 *
+	 * @param array $array The array to add to.
+	 * @param mixed $value The value to add.
+	 *
+	 * @return array
+	 */
+	private function add_to_array_unique( $array, $value ) {
```

Comment:
@preda-bogdan what do you think instead of using a special method to check this to have a key based array for checking this, i.e 
```php
$original_post_ids[$original_post_id] = true
```
And check for existence with isset.

This should be more performant in this case.

Reply:
@selul thank you for the suggestion. I agree it should perform better.